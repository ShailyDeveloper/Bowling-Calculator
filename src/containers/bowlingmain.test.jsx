import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import BowlHandler from './bowlingmain';
import '@testing-library/jest-dom';

configure({ adapter: new Adapter() });

describe('DisplayControl', () => {
    it('should not render the button at the launch');
    const wrapper = shallow(<BowlHandler></BowlHandler>);
    expect(wrapper.find(BowlHandler)).toMatchSnapshot();

});