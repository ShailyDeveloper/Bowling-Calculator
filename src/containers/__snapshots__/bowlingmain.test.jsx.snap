// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): ShallowWrapper {
    Symbol(enzyme.__root__): [Circular],
    Symbol(enzyme.__unrendered__): <Bowling />,
    Symbol(enzyme.__renderer__): Object {
      "batchedUpdates": [Function],
      "checkPropTypes": [Function],
      "getNode": [Function],
      "render": [Function],
      "simulateError": [Function],
      "simulateEvent": [Function],
      "unmount": [Function],
    },
    Symbol(enzyme.__node__): Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            style={
                        Object {
                                    "margin": "10px",
                                  }
            }
>
            Bowling
</h1>,
          <BowlHeaders
            buttonhandler={[Function]}
            finalroundhandle={[Function]}
            finalroundspare={[Function]}
            handleChanged={[Function]}
            showsecond={
                        Object {
                                    "lastspare": 0,
                                    "laststrike": 0,
                                    "roundcount": 1,
                                    "score": 0,
                                    "showval": null,
                                    "spareanddoublestrike": 0,
                                    "str1Val": null,
                                    "str2Val": null,
                                    "strike": 0,
                                  }
            }
/>,
          <br />,
          <br />,
          <label
            style={
                        Object {
                                    "margin": "10px",
                                  }
            }
>
            <b>
                        Score
            </b>
</label>,
          " ",
          <label>
            <b />
</label>,
          <label
            style={
                        Object {
                                    "margin": "10px",
                                  }
            }
>
            <b />
</label>,
          " ",
          <label>
            <b />
</label>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Bowling",
            "style": Object {
              "margin": "10px",
            },
          },
          "ref": null,
          "rendered": "Bowling",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "buttonhandler": [Function],
            "finalroundhandle": [Function],
            "finalroundspare": [Function],
            "handleChanged": [Function],
            "showsecond": Object {
              "lastspare": 0,
              "laststrike": 0,
              "roundcount": 1,
              "score": 0,
              "showval": null,
              "spareanddoublestrike": 0,
              "str1Val": null,
              "str2Val": null,
              "strike": 0,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <b>
              Score
</b>,
            "style": Object {
              "margin": "10px",
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Score",
            },
            "ref": null,
            "rendered": "Score",
            "type": "b",
          },
          "type": "label",
        },
        " ",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <b />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "b",
          },
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <b />,
            "style": Object {
              "margin": "10px",
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": 0,
            },
            "ref": null,
            "rendered": 0,
            "type": "b",
          },
          "type": "label",
        },
        " ",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <b />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "b",
          },
          "type": "label",
        },
      ],
      "type": "div",
    },
    Symbol(enzyme.__nodes__): Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1
              style={
                            Object {
                                          "margin": "10px",
                                        }
              }
>
              Bowling
</h1>,
            <BowlHeaders
              buttonhandler={[Function]}
              finalroundhandle={[Function]}
              finalroundspare={[Function]}
              handleChanged={[Function]}
              showsecond={
                            Object {
                                          "lastspare": 0,
                                          "laststrike": 0,
                                          "roundcount": 1,
                                          "score": 0,
                                          "showval": null,
                                          "spareanddoublestrike": 0,
                                          "str1Val": null,
                                          "str2Val": null,
                                          "strike": 0,
                                        }
              }
/>,
            <br />,
            <br />,
            <label
              style={
                            Object {
                                          "margin": "10px",
                                        }
              }
>
              <b>
                            Score
              </b>
</label>,
            " ",
            <label>
              <b />
</label>,
            <label
              style={
                            Object {
                                          "margin": "10px",
                                        }
              }
>
              <b />
</label>,
            " ",
            <label>
              <b />
</label>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Bowling",
              "style": Object {
                "margin": "10px",
              },
            },
            "ref": null,
            "rendered": "Bowling",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "buttonhandler": [Function],
              "finalroundhandle": [Function],
              "finalroundspare": [Function],
              "handleChanged": [Function],
              "showsecond": Object {
                "lastspare": 0,
                "laststrike": 0,
                "roundcount": 1,
                "score": 0,
                "showval": null,
                "spareanddoublestrike": 0,
                "str1Val": null,
                "str2Val": null,
                "strike": 0,
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <b>
                Score
</b>,
              "style": Object {
                "margin": "10px",
              },
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Score",
              },
              "ref": null,
              "rendered": "Score",
              "type": "b",
            },
            "type": "label",
          },
          " ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <b />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "b",
            },
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <b />,
              "style": Object {
                "margin": "10px",
              },
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": 0,
              },
              "ref": null,
              "rendered": 0,
              "type": "b",
            },
            "type": "label",
          },
          " ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <b />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "b",
            },
            "type": "label",
          },
        ],
        "type": "div",
      },
    ],
    Symbol(enzyme.__options__): Object {
      "adapter": ReactSixteenAdapter {
        "options": Object {
          "enableComponentDidUpdateOnSetState": true,
          "legacyContextMode": "parent",
          "lifecycles": Object {
            "componentDidUpdate": Object {
              "onSetState": true,
            },
            "getChildContext": Object {
              "calledByRenderer": false,
            },
            "getDerivedStateFromError": true,
            "getDerivedStateFromProps": Object {
              "hasShouldComponentUpdateBug": false,
            },
            "getSnapshotBeforeUpdate": true,
            "setState": Object {
              "skipsComponentDidUpdateOnNullish": true,
            },
          },
        },
      },
      Symbol(enzyme.__providerValues__): undefined,
    },
    Symbol(enzyme.__providerValues__): Map {},
    Symbol(enzyme.__childContext__): null,
  },
  Symbol(enzyme.__unrendered__): null,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): undefined,
  Symbol(enzyme.__nodes__): Array [],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__rootNodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            style={
                        Object {
                                    "margin": "10px",
                                  }
            }
>
            Bowling
</h1>,
          <BowlHeaders
            buttonhandler={[Function]}
            finalroundhandle={[Function]}
            finalroundspare={[Function]}
            handleChanged={[Function]}
            showsecond={
                        Object {
                                    "lastspare": 0,
                                    "laststrike": 0,
                                    "roundcount": 1,
                                    "score": 0,
                                    "showval": null,
                                    "spareanddoublestrike": 0,
                                    "str1Val": null,
                                    "str2Val": null,
                                    "strike": 0,
                                  }
            }
/>,
          <br />,
          <br />,
          <label
            style={
                        Object {
                                    "margin": "10px",
                                  }
            }
>
            <b>
                        Score
            </b>
</label>,
          " ",
          <label>
            <b />
</label>,
          <label
            style={
                        Object {
                                    "margin": "10px",
                                  }
            }
>
            <b />
</label>,
          " ",
          <label>
            <b />
</label>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Bowling",
            "style": Object {
              "margin": "10px",
            },
          },
          "ref": null,
          "rendered": "Bowling",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "buttonhandler": [Function],
            "finalroundhandle": [Function],
            "finalroundspare": [Function],
            "handleChanged": [Function],
            "showsecond": Object {
              "lastspare": 0,
              "laststrike": 0,
              "roundcount": 1,
              "score": 0,
              "showval": null,
              "spareanddoublestrike": 0,
              "str1Val": null,
              "str2Val": null,
              "strike": 0,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <b>
              Score
</b>,
            "style": Object {
              "margin": "10px",
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Score",
            },
            "ref": null,
            "rendered": "Score",
            "type": "b",
          },
          "type": "label",
        },
        " ",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <b />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "b",
          },
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <b />,
            "style": Object {
              "margin": "10px",
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": 0,
            },
            "ref": null,
            "rendered": 0,
            "type": "b",
          },
          "type": "label",
        },
        " ",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <b />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "b",
          },
          "type": "label",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__providerValues__): null,
}
`;
